coral-trino/src/main/java/com/linkedin/coral/trino/rel2trino/utils/CoralToTrinoSqlCallTransformersUtil.java

```diff
@@ -0,0 +1,166 @@
+    DEFAULT_SQL_CALL_TRANSFORMER_LIST.add(createSignatureBasedConditionSqlCallTransformer(
+        hiveToCoralSqlOperator("pmod"), 2, "mod",
+        "[{\"op\":\"+\",\"operands\":[{\"op\":\"%\",\"operands\":[{\"input\":1},{\"input\":2}]},{\"input\":2}]},{\"input\":2}]",
+        null, null));
+    DEFAULT_SQL_CALL_TRANSFORMER_LIST
+        .add(createSignatureBasedConditionSqlCallTransformer(hiveToCoralSqlOperator("base64"), 1, "to_base64"));
+    DEFAULT_SQL_CALL_TRANSFORMER_LIST
+        .add(createSignatureBasedConditionSqlCallTransformer(hiveToCoralSqlOperator("unbase64"), 1, "from_base64"));
+    DEFAULT_SQL_CALL_TRANSFORMER_LIST
+        .add(createSignatureBasedConditionSqlCallTransformer(hiveToCoralSqlOperator("hex"), 1, "to_hex"));
+    DEFAULT_SQL_CALL_TRANSFORMER_LIST
+        .add(createSignatureBasedConditionSqlCallTransformer(hiveToCoralSqlOperator("unhex"), 1, "from_hex"));
+    DEFAULT_SQL_CALL_TRANSFORMER_LIST
+        .add(createSignatureBasedConditionSqlCallTransformer(hiveToCoralSqlOperator("array_contains"), 2, "contains"));
+    DEFAULT_SQL_CALL_TRANSFORMER_LIST.add(
+        createSignatureBasedConditionSqlCallTransformer(hiveToCoralSqlOperator("regexp_extract"), 3, "regexp_extract",
+            "[{\"input\": 1}, {\"op\": \"hive_pattern_to_trino\", \"operands\":[{\"input\": 2}]}, {\"input\": 3}]",
+            null, null));
+    DEFAULT_SQL_CALL_TRANSFORMER_LIST
+        .add(createSignatureBasedConditionSqlCallTransformer(hiveToCoralSqlOperator("instr"), 2, "strpos"));
+    DEFAULT_SQL_CALL_TRANSFORMER_LIST
+        .add(createSignatureBasedConditionSqlCallTransformer(hiveToCoralSqlOperator("decode"), 2,
+            "[{\"regex\":\"(?i)('utf-8')\", \"input\":2, \"name\":\"from_utf8\"}]", "[{\"input\":1}]", null, null));
+
+    DEFAULT_SQL_CALL_TRANSFORMER_LIST
+        .add(createSignatureBasedConditionSqlCallTransformer(hiveToCoralSqlOperator("date_add"), 2, "date_add",
+            "[{\"value\": 'day'}, {\"input\": 2},  "
+                + "{\"op\": \"date\", \"operands\":[{\"op\": \"timestamp\", \"operands\":[{\"input\": 1}]}]}]",
+            null, null));
+    DEFAULT_SQL_CALL_TRANSFORMER_LIST
+        .add(createSignatureBasedConditionSqlCallTransformer(hiveToCoralSqlOperator("date_sub"), 2, "date_add",
+            "[{\"value\": 'day'}, " + "{\"op\": \"*\", \"operands\":[{\"input\": 2}, {\"value\": -1}]}, "
+                + "{\"op\": \"date\", \"operands\":[{\"op\": \"timestamp\", \"operands\":[{\"input\": 1}]}]}]",
+            null, null));
+    DEFAULT_SQL_CALL_TRANSFORMER_LIST.add(createSignatureBasedConditionSqlCallTransformer(
+        hiveToCoralSqlOperator("datediff"), 2, "date_diff",
+        "[{\"value\": 'day'}, {\"op\": \"date\", \"operands\":[{\"op\": \"timestamp\", \"operands\":[{\"input\": 2}]}]}, "
+            + "{\"op\": \"date\", \"operands\":[{\"op\": \"timestamp\", \"operands\":[{\"input\": 1}]}]}]",
+        null, null));
+  }
+
+  private static void addLinkedInFunctionTransformers() {
```

Comment:
Can LinkedIn transformers be a subclass of `SignatureBasedConditionSqlCallTransformer`>

Reply:
I don't think it is necessary. LnkedIn transformers just have the value of the field of `fromOperatorName` started with `com.linkedin` and their behaviors (including the implementation of `condition` and `transform` functions) are the same as other `OperatorBasedSqlCallTransformer` (namely `SignatureBasedConditionSqlCallTransformer`) before.

Reply:
Creating a new class named `LinkedInOperatorBasedSqlCallTransformer` which is a subclass of `OperatorBasedSqlCallTransformer` to transforms the function operator from Coral to Trino on SqlNode layer.